### ISSUE TITLE
Followup TODO list for FAKE 5 before a real release

### ISSUE THREAD

**Issue by matthid** [fsharp/FAKE#1523]
As the older issues https://github.com/fsharp/FAKE/pull/1281 and https://github.com/fsharp/FAKE/issues/1232 are quite large due to discussion this is a new issue to track the progress/todo list before we can issue a FAKE 5 release. I think we are ready for an alpha release at this point.

TODO (Release)

- [ ] Add timing information for the new features (paket resolution & download) -> https://github.com/fsharp/FAKE/issues/1831
- [ ] Target command line arguments? -> https://github.com/fsharp/FAKE/issues/1830
- [x] various BuildServer Integrations?... with https://github.com/fsharp/FAKE/pull/1819
- [x] public beta phase and integration in at least 3 other projects?
       - https://github.com/fsprojects/FSharp.Formatting/pull/460
       - https://github.com/isaacabraham/vsts-fsharp
       - https://github.com/fsprojects/Mechanic/pull/90
- [x] quiet mode to make scripting and piping possible?
- [x] PERFORMANCE Warmstart (valid lockfile + valid assembly cache) MUST be faster
       -> Performance is acceptable now
- [x] Need a feature to update packages when using the "in-line" header. -> https://github.com/fsharp/FAKE/issues/1769
- [x] Write documentation for migrating to FAKE 5 (Installing)
- [x] Write documentation for installing FAKE and all the install options
- [x] When a target fails it seems to be missing from the build overview/summary...
- [x] implement load script generation (using paket) -> for tooling support.
- [x] fsharpformatting support
- [x] Fix AUTHOR in dotnetcore packages...
- [x] [Write API Guidelines (casing using, ...) (API)](https://fake.build/contributing.html#API-Design)
- [x] rename "Fake" -> "fake"
- [x] fake website at [fake.build](https://fake.build)
- [x] Better defaults: Use a comment in the `paket.dependencies` file when no FAKE header exists (thanks @cloudRoutine) (https://github.com/matthid/FAKE/pull/7)
- [x] ~~Remove duplicate code from FakeLib (mutables!), not sure if this is still possible with the new context concept.~~ -> The old code is obsolete and will no longer be touched
- [x] Remove binaries from lib folder (currently these "special" version are required)
- [x] Make the caching layer in fake load the correct assembly (depending on the current runtime and the runtime folder of the packages) if necessary (Microsoft.NETCore.Platforms). Paket support was added via https://github.com/fsprojects/Paket/pull/2255
- [x] Fake.Tools.Git (such that we can release FAKE 5 with FAKE 5 :)
- [x] Remove warning when we load an assembly (when PublicToken is null)
- [x] Rename "Fake.netcore.exe" to "Fake.exe"
- [x] Remove lockfile outdated check and warning
- [x] Write documentation for [migrating to FAKE 5 (CLI)](https://github.com/matthid/FAKE/blob/coreclr/help/migrate-to-fake-5.md)
- [x] Docs for [CommandLine](https://github.com/matthid/FAKE/blob/coreclr/help/commandline.md)
- [x] Write documentation for [migrating to FAKE 5 (API)](https://github.com/matthid/FAKE/blob/coreclr/help/migrate-to-fake-5.md)
- [x] Improve error messages when using FAKE 5 netcore version as drop-in replacement (thanks @cloudRoutine)
- [x] Make a chocolatey package and document it (https://chocolatey.org/packages/fake/ further docs will be part of the install options)

IF CONTRIBUTED (otherwise obsolete with release)
- [ ] Improve obsolete warnings and suggest a paket goup (help API migration, thanks @cloudRoutine)

TODO (Later) -> might be moved to separate issues later
- [ ] https://github.com/jordansissel/fpm (basically have a way to install on unixes) -> See https://github.com/fsharp/FAKE/issues/1828
- [ ] Trim packages ([here](https://docs.microsoft.com/de-de/dotnet/articles/core/deploying/reducing-dependencies) and [here](https://docs.microsoft.com/de-de/dotnet/articles/core/tutorials/managing-package-dependency-versions)) -> See https://github.com/fsharp/FAKE/issues/1829
- [x] ~~Add command to generate/update `fake.sh` and `fake.cmd` and delete `obtain_fake.sh`.~~ Maybe later. Lets see how many people use this way of installing FAKE.
- [x] Add the `fake` coreclr tool -> maybe bootstrapping is better? Lets wait until dotnet cli supports global tools? UP-FOR-GRAPS
- [ ] ~~Add `--interactive` flag to run scripts interactively? -> Decide if we still need that. The only use cases are dependencies, which are now managed by paket/fake... Feedback is welcome~~ -> I'm pretty happy with how it works right now
- [ ] ~~Error `**** error: internal error: The type initializer for '<StartupCode$FSharp-Compiler-Service-netcore>.$Reshapedmsbuild' threw an exception.` when running with `fsi` object support~~ -> We currently use the compiler because fsi is not working on netcore (we only need that if we decide that we need `--interactive`)

**Comment by matthid** [comment]
Closing this as almost everything is done, only two points are missing and I created new issues for them
